src/GraphQL.Primitives/Hash.cs

```diff
@@ -0,0 +1,44 @@
+using System.Buffers;
+using System.Diagnostics;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace GraphQL;
+
+internal static class Hash
+{
+    private static SHA256? _sha256;
+
+    internal static string Compute(string query)
+    {
+        int expected = Encoding.UTF8.GetByteCount(query);
+        byte[]? inputBytes = ArrayPool<byte>.Shared.Rent(expected);
+        int written = Encoding.UTF8.GetBytes(query, 0, query.Length, inputBytes, 0);
```

Comment:
This could be rewritten to use Span / stackalloc also.

Reply:
Could you please elaborate? 😉

Reply:
I saw that stackalloc was used on lines 21-27.  Stackalloc could be used for the call to GetBytes as well.  This is faster than ArrayPool.

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/stackalloc

The easy recommended pattern is to write code like this:

```cs
int length = 1000;
Span<byte> buffer = length <= 1024 ? stackalloc byte[length] : new byte[length];
```

This assumes that typically the buffer length is relatively small, and in such cases, is ideal.  It also includes a fallback for large buffer requirements; but then you'd probably lose the ArrayPool (?).  And if it was known that the query would commonly be larger than you'd want to allocate on the stack, then maybe the cost of the heap allocation for the long queries might be worse than the benefits from stackalloc.  Or you'd have to write a separate if block for the ArrayPool path vs not.  I'm not really sure; maybe better off leaving it as-is.

Another possible optimization would be to guess at the length, rather than using GetByteCount.  You'd have to write a fallback routine in case the encoded string length was longer than the preallocated buffer though, and you might need to use the underlying Encoder rather than the static method.

Reply:
I think I'll leave it as it is. Using `Span<byte>` would require separate code for old frameworks again, and the best optimization is using the `GraphQLQuery` class to compute the hash only once... but anyway, thanks for the explanation!